test/sqllogictest/transform/predicate_pushdown.slt

```diff
@@ -441,6 +441,8 @@ Explained Query:
               Map ("materialized-view") // { arity: 11 }
                 ReadIndex on=mz_materialized_views mz_materialized_views_ind=[*** full scan ***] // { arity: 10 }
 
+Source mz_internal.mz_continual_tasks
```

Comment:
Hmm, this seems wrong?

Reply:
It used to be like this, but this was super confusing, so [we changed this to print sources even if they don't have any MFP](https://materializeinc.slack.com/archives/CM7ATT65S/p1716542196064999?thread_ts=1716497300.875229&cid=CM7ATT65S) in 2024 May.

However, there are two more confusing things around here (which are unfortunately still present even after the above change):

1.

When all Gets to a source are optimized out, the source disappears from EXPLAIN _only_ if there was an _indexed_ read to it. So, what is happening here is that:
- The Gets to `mz_tables` and `mz_sources` are disappearing due to optimizations. Since these are indexed reads, they also disappear from EXPLAIN.
- The Get to `mz_continual_tasks` is also disappearing due to optimizations. However, EXPLAIN still prints it, because it's a non-indexed read (it's a Persist read). (However, we are not actually going to read it, due to how rendering and/or Timely dataflow works. I think, but maybe @frankmcsherry can confirm.)

So, I think there isn't any big issue here apart from EXPLAIN, for which I'll open an issue shortly.

But one thing that I'd suggest to still do in this PR is to add a similar index to `mz_continual_tasks` as the indexes on `mz_tables` and `mz_sources`, for symmetry.

2.

The relation between the sources list and the indexes list is not clear to a non-expert reader of EXPLAIN. We should either
- Change formatting slightly to indicate that the sources list is showing storage reads (i.e., that that list never includes things that will also appear in the indexes list), e.g., prefix that list with "Storage Reads:" and change the "Used indexes" to "Index Reads:". (This is what I suggested [here](https://materializeinc.slack.com/archives/CM7ATT65S/p1716542645960309?thread_ts=1716497300.875229&cid=CM7ATT65S), but unfortunately it didn't happen in that EXPLAIN change.)
- Or just merge the two lists into a single "Sources" list, and indicate for each source whether it's gonna be a storage read or index read.

Reply:
Cool, that all makes sense. Thanks both!

> But one thing that I'd suggest to still do in this PR is to add a similar index to mz_continual_tasks as the indexes on mz_tables and mz_sources, for symmetry.

Done!